import "goby/core/proto/option_extensions.proto";

message AppBaseConfig
{
  required string platform_name = 1 [(example)="AUV-23",
                                     (description) = "same as self.name for gobyd cfg"];
  
  optional string app_name = 2 [(example)="myapp_g",
                                (description) = "default is compiled name - change this to run multiple instances"];

  enum Verbosity
  {
    quiet = 1;
    warn = 2;
    verbose = 3;
    debug = 4;
    gui = 5;
  }
  
  optional Verbosity verbosity = 3 [default = quiet,
                                    (description)="Terminal verbosity"];    


  // used only by ApplicationBase, ignored by gobyd
  optional float loop_freq = 100 [default = 10,
                                  (description) = "the frequency (Hz) used to run loop()"]; 
}

